### PR TITLE
Migrate JS to TS with npm

### PR DESCRIPTION
**Pull request by ntsagar9**
**Description**

- Migrate JS to TS with added NPM
- Implemented linting for clean code
- Update README.md file

### CODE REVIEW

File: src/type.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface FlagsObjectInterface {
+    [key: string]: string
```
Comment (imabp) [inline src/type.ts]: The type should be `[key:string]: any`, 

Think of command like
`node index.js -f <FILE>`
`node index.js -f <OBJECT>`

While reading from CLI, objects will be stringified, but we need to return it as an object literal so consumers can use it.

Hence, `[key:string]:any` is preferred.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "flagparser",
+  "version": "1.0.0",
```
Comment (imabp) [inline package.json]: version should be `0.0.1`,  as this is not yet stable to use.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./node_modules/gts/tsconfig-google.json",
+  "compilerOptions": {
+    "rootDir": ".",
```
Comment (imabp) [inline tsconfig.json]: `rootDir` should be `src/*` as we are having `src/index.ts` our entry file.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "description": "We are building a feature-rich flag parser.",
+  "main": "index.js",
```
Comment (imabp) [inline package.json]: `main` should be referred from the tsConfig `outDir` key, hence it should be `build/index.tsx`

File: package.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "parse command line flags"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/imabp/flagparser.git"
+  },
+  "engine": {
+    "node": ">=6.0.0"
+  },
+  "author": "ntsagar9@gmail.com",
```
Comment (imabp) [inline package.json]: we should remove author for now. I will add this when we publish the package to npm.

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    "outDir": "build"
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts"
```
Comment (imabp) [inline tsconfig.json]: replace `test` with `__tests__`

**Review by imabp — CHANGES_REQUESTED** [review]
Nice work @ntsagar9  . There are some suggestions, please check them out in the review.

File: package.json (hunk 4 above)
Comment (ntsagar9) [reply package.json]: main should be `build/index.js`

**Review by imabp — APPROVED** [review]
LGTM!! Approved

**Pull request merged**